pkg/reconciler/pipelinerun/cancel.go

```diff
@@ -73,6 +74,21 @@ func init() {
 
 func cancelRun(ctx context.Context, runName string, namespace string, clientSet clientset.Interface) error {
 	_, err := clientSet.TektonV1alpha1().Runs(namespace).Patch(ctx, runName, types.JSONPatchType, cancelRunPatchBytes, metav1.PatchOptions{}, "")
+	if errors.IsNotFound(err) {
```

Comment:
Should we instead first check if the resource exists before cancelling it?

Reply:
I personally prefer just ignoring not found errors - it keeps the code simpler and achieves the same end result.

Reply:
I agree but that causes unnecessary error showing up in the logs which can be confusing. I would rather avoid such error which will be ignored later by the same function

Reply:
I don't think it would show any error - it just returns nil if the `TaskRun` or `Run` doesn't exist.